### DISCUSSION TITLE
CORS handling with zitadel.cloud not aware of addiitional origins settings?

### DISCUSSION THREAD

**Discussion by SBejga** [zitadel/zitadel#5000] [Q&A]
Hello,

I am developing a web app (PWA) and want to use Zitadel as backend for authentication.

I use https://github.com/moberwasserlechner/capacitor-oauth2 as a framework.
I have a Code flow and in the logs I see the auth code is okay, but when trying to exchange to access token, I am getting the following errors:

> Access to XMLHttpRequest at 'https://<my-instance>.zitadel.cloud/oauth/v2/token' from origin 'http://localhost:8100' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

And I am wondering, because I added the origin `http://localhost:8100` to the project application at "REDIRECT SETTINGS" and additionally at "ADDITIONAL ORIGINS".  I assumed, that it should allow CORS then for these origins? Additionally, the development mode is active.

Any clue or hints?

Best regards,
Sebastian

**Comment by livio-a** [comment]
Hey @SBejga 

We discovered an issue in our CORS (preflight) response on friday: https://github.com/zitadel/zitadel/issues/4995
Please subscribe to this issue, to track it's progress.
I'll make sure we prioritise this as it's impacting more and more developers.

BTW if the redirect_uri already contains `http://localhost:8100`, you won't need to provide an additional origin. We build and allow the origins of all provided redirect_uris. Additional origins are useful / needed if you have for example a react-native application, where your redirect_uri is a custom scheme (e.g. my.app://callback) but you need to allow http://locahost for the app to work.

Best regards
Livio

**Reply by kneoio** [reply]
Hi. In the end how to avoid the CORS issue? It looks like developing an app locally with the server is a little bit tricky when I use Docker container with Zitadel. I just wonder how the developer has done it when the test.  Sure, I can clone,  deploy, and tweak it locally but it looks a bit cumbersome

**Reply by hifabienne** [reply]
Can you please tell, what exactly your problem is? What are you trying to achieve?
The issue we had above, was already fixed.

**Reply by nishad-bdg** [reply]
I am also getting the same issue in the localhost.  
![Screenshot 2024-04-19 at 12 10 52 PM](https://github.com/zitadel/zitadel/assets/3539147/9b9c7d19-4df3-4d42-bd31-cdb8a7259be8)
